compiler/luci/service/src/Nodes/CirclePad.test.cpp

```diff
@@ -31,3 +33,59 @@ TEST(CloneNodeTest, clone_Pad)
   auto cloned_pad = dynamic_cast<luci::CirclePad *>(cloned);
   ASSERT_NE(nullptr, cloned_pad);
 }
+
+TEST(ShapeRuleTest, pad_dynamic_shape)
+{
+  luci::CirclePad pad;
+  luci::CircleInput input;
+  luci::CircleConst padddings;
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  input.shape({1, 2, 3, 4});
+  input.shape_status(luci::ShapeStatus::VALID);
+  input.dim(2).unset();
+
+  padddings.dtype(loco::DataType::S64);
+  padddings.shape({4, 2});
+  padddings.shape_status(luci::ShapeStatus::VALID);
+
+  const loco::DataType S64 = loco::DataType::S64;
+  uint32_t t = 64 * 8;
+  padddings.size<S64>(t);
+
+  pad.input(&input);
+  pad.paddings(&padddings);
+
+  ASSERT_TRUE(shape_inf_rule.infer(&pad, shape));
+  ASSERT_EQ(shape.rank(), 4);
+  ASSERT_TRUE(shape.dim(0).known());
+  ASSERT_TRUE(shape.dim(1).known());
+  ASSERT_FALSE(shape.dim(2).known());
+  ASSERT_TRUE(shape.dim(3).known());
+
+  ASSERT_EQ(1, shape.dim(0).value());
+  ASSERT_EQ(2, shape.dim(1).value());
+  ASSERT_EQ(0, shape.dim(2).value());
+  ASSERT_EQ(4, shape.dim(3).value());
+  pad.drop();
```

Comment:
maybe we can fix this problem using graph.. (but not sure)
```cxx
  auto g = loco::make_graph();
  auto node_pad = g->nodes()->create<luci::CirclePad>();
```
and so on... or

```suggestion
  // to prevent "Assertion '_uses.size() == 0'" failure
  pad.drop();
```

Reply:
> maybe we can fix this problem using graph.. (but not sure)
> 
> ```c++
>   auto g = loco::make_graph();
>   auto node_pad = g->nodes()->create<luci::CirclePad>();
> ```

By using graph, `.drop()` is no more needed since when graph is destroyed, it explicitly remove node_pool. (@zetwhite explained to me.)
https://github.com/Samsung/ONE/blob/987fafaa0b21023f8e6cb93aa9e8b586fc6a123a/compiler/loco/include/loco/IR/Graph.h#L226-L228
https://github.com/Samsung/ONE/blob/987fafaa0b21023f8e6cb93aa9e8b586fc6a123a/compiler/loco/src/IR/NodePool.cpp#L25-L28

Let's use `graph`, then.